File: Day10_FileIO/main.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
 
public class Main 
{   
    public static void main(String[] args) 
    {
        BufferedReader reader = null;
        int charCount = 0;
        int wordCount = 0;
        int lineCount = 0;
        
        reader = new BufferedReader(new FileReader("Input.txt"));
        String currentLine = reader.readLine();
             
        while (currentLine != null)
            {
                lineCount++;
                String[] words = currentLine.split(" ");
                
                wordCount+= words.length;
                for (String word : words)
                {
                    charCount += word.length();
                }
                currentLine = reader.readLine();
            }
            
            System.out.println("Chars "+charCount);
            System.out.println("Words "+wordCount);
            System.out.println("Lines "+lineCount); 
    }
}
        

File: Day10_FileIO/main.py
import os

def compute(fname):
    words = 0
    lines = 0
    chars = 0
    spaces = 0
    with open(fname,'r') as file:
        for line in file:
            line = line.strip(os.linesep)
            wordList = line.split()
            lines +=1
            words += len(wordList)
            chars += sum(1 for c in line
                         if c not in(os.linesep,' '))
            spaces += sum(1 for s in line
                          if s in (os.linesep,' '))
    print("words :",words)
    print("lines :",lines)
    print("characters:",chars)
    print("spaces: ",spaces)
    
if __name__ == '__main__':
    fname = "Input.txt"
    try:
        compute(fname)
    except:
        print('File not found.')    
    
